geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/RegionAdvisor.java

```diff
@@ -180,19 +180,23 @@ public void processProfilesQueuedDuringInitialization() {
               logger.trace(LogMarker.DISTRIBUTION_ADVISOR_VERBOSE,
                   "applying queued profile removal for all buckets for {}", qbp.memberId);
             }
-            for (int i = 0; i < buckets.length; i++) {
-              BucketAdvisor ba = buckets[i].getBucketAdvisor();
-              int serial = qbp.serials[i];
-              if (serial != ILLEGAL_SERIAL) {
-                ba.removeIdWithSerial(qbp.memberId, serial, qbp.destroyed);
-              }
-            } // for
+            if (qbp.serials.length == 1) {
```

Comment:
it looks the "if" can be handled in "else"

Reply:
Looking at the code again, when `serials.length == 1`, the `bucketId` might not necessarily be `0`. Therefore, we still need the `if` and `else`.